### DISCUSSION TITLE
Latest FreeCAD (1.0.0) breaks KiCad import script

### DISCUSSION THREAD

**Discussion by WallieEverest** [LubomirJagos42/FreeCAD-OpenEMS-Export#10] [General]
The `KiCad PCB Import Tool` button is known to be disabled when first launching the macro. This is remedied by relaunching the macro. The fix works up through FreeCAD 0.21.2. Unfortunately, FreeCAD 1.0.0 is behaving differently, and the _import button never becomes enabled_.
{Bug fix request}
Tested configuration is Windows 11, Python 3.11, and both portable versions (.7z) of FreeCAD.
https://github.com/FreeCAD/FreeCAD/releases
* FreeCAD_1.0.0-conda-Windows-x86_64-py311.7z
* FreeCAD-0.21.2-Windows-x86_64.7z
Note: I work on a company-managed machine and do not have admin rights by default. This may impact how Python libraries (PySide) are parsed. This configuration is becoming more common for users and may not reflect the environment where open-source testing is typically performed. For example, the admin/user PATH environments often need consideration after a software install.

**Comment by krissn0** [comment]
I see WebGui is not available. Therefore this fails.
```
try:
	import FreeCAD
	import WebGui
	import KiCADImporterToolDialog	#import for KiCAD Import Tool
	APP_CONTEXT = "FreeCAD"
```

ps. my knowledge of python is 1/10 so take this into account

**Reply by krissn0** [reply]
commenting ` import WebGui` solve the issue. This is only use to open help page.

**Comment by WallieEverest** [comment]
Confirmed this suggested change patches Python macro to continue working with FreeCAD 1.0.0.

File: ExportOpenEMSDialog.py
*** Line changes ***
17:  #WE 2024-12-06  import WebGui # does not exist in FreeCAD 1.0.0
520:  #WE 2024-12-06  WebGui.openBrowser(f"{os.path.dirname(__file__)}\\documentation\\help\\index.html")

**Comment by kohoutm2** [comment]
I fixed the issue by replacing
import WebGui
and
WebGui.openBrowser(f"{os.path.dirname(__file__)}\\documentation\\help\\index.html")

by

import webbrowser
and
webbrowser.open(f"{os.path.dirname(__file__)}\\documentation\\help\\index.html")

**Reply by krissn0** [reply]
should be rather:
`webbrowser.open(f"{os.path.dirname(__file__)}\\documentation\\help\\index.html")`

**Reply by james7x1** [reply]
Excellent, worked like a charm. Didn't see a PR for this.

**Comment by 9662e103-129a** [comment]
There also seems to be some issue saving grid settings (FCAD 1.0.2 and latest version of this tool). I submitted [this issue](https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/issues/55) to the original repo issue section.
